pyproject.toml

```
@@ -93,3 +93,61 @@ environment = { MACOSX_DEPLOYMENT_TARGET = "10.15", DEPLOY = "ON" }
 [tool.cibuildwheel.windows]
 before-build = "pip install delvewheel"
 repair-wheel-command = "delvewheel repair -v -w {dest_dir} {wheel}"
+
+
+[tool.ruff]
+select = [
+    "E", "F", "W", # flake8
+    "A",           # flake8-builtins
+    "B",  "B904",  # flake8-bugbear
+    "I",           # isort
+    "ARG",         # flake8-unused-arguments
+    "C4",          # flake8-comprehensions
+    "EM",          # flake8-errmsg
+    "EXE",         # flake8-executable
+    "ICN",         # flake8-import-conventions
+    "ISC",         # flake8-implicit-str-concat
+    "PGH",         # pygrep-hooks
+    "PIE",         # flake8-pie
+    "PL",          # pylint
+    "PT",          # flake8-pytest-style
+    "PTH",         # flake8-use-pathlib
+    "Q",           # flake8-quotes
+    "RET",         # flake8-return
+    "RUF",         # Ruff-specific
+    "SIM",         # flake8-simplify
+    "TCH",         # flake8-type-checking
+    "UP",          # pyupgrade
+    "YTT",         # flake8-2020
+]
+ignore = ["E501"]
```

We could just add `"PLR2004", "PLR0913",` here as well. (magic numbers and too many arguments)
Just looked into other repositories and these two rules seem to be ignored in most of them.
```suggestion
ignore = [
  "PLR2004", # Magic values
  "PLR0913", # Too many arguments
  "E501",    # Line too long (Black is enough)
]
```